packages/edit-site/src/components/editor/index.js

```diff
@@ -50,6 +50,39 @@ const interfaceLabels = {
 	footer: __( 'Editor footer' ),
 };
 
+function useIsSiteEditorLoading() {
+	const { isLoaded: hasLoadedPost } = useEditedEntityRecord();
+	const { hasResolvingSelectors } = useSelect( ( select ) => {
+		return {
+			hasResolvingSelectors: select( coreStore ).hasResolvingSelectors(),
+		};
+	} );
```

Comment:
One more optimization opportunity: this `useSelect` will return a new value every time any selector starts or stops resolving, even long after the editor is fully loaded. That will trigger a rerender of the `Editor` component that uses the `useIsSiteEditorLoading` hook.

You can make the return value more stable by including the `loaded` value as dependency:
```js
const [ loaded, setLoaded ] = useState( false );
const inLoadingPause = useSelect( ( select ) => {
  const hasResolvingSelectors = select( coreStore ).hasResolvingSelectors();
  return ! loaded && ! hasResolvingSelectors;
}, [ loaded ] );
```
This will guarantee a stable `false` return value once the `loaded` state is achieved. No more renderers.

Reply:
Good catch, I'll follow up in another PR.

Reply:
See #50546